plenum/server/batch_handlers/audit_batch_handler.py

```diff
@@ -108,23 +108,32 @@ def _create_audit_txn_data(self, three_pc_batch, last_audit_txn):
 
         return txn
 
-    def __fill_ledger_root_hash(self, txn, three_pc_batch, lid, last_audit_txn):
-        target_ledger_id = three_pc_batch.ledger_id
+    def __fill_ledger_root_hash(self, txn, lid, ledger, last_audit_txn):
         last_audit_txn_data = get_payload_data(last_audit_txn) if last_audit_txn is not None else None
 
-        # 1. ledger is changed in this batch => root_hash
-        if lid == target_ledger_id:
-            txn[AUDIT_TXN_LEDGER_ROOT][lid] = Ledger.hashToStr(three_pc_batch.txn_root)
+        # 1. it is the first batch and we have something
+        if last_audit_txn_data is None and ledger.uncommitted_size:
+            txn[AUDIT_TXN_LEDGER_ROOT][lid] = Ledger.hashToStr(ledger.uncommitted_root_hash)
 
-        # 2. This ledger is never audited, so do not add the key
+        # 1.1. Rare case -- we have previous audit txns but don't have this ledger i.e. new plugins
```

Comment:
Please add unit tests for each of this cases

Reply:
Enhanced old unit test and added a new one for the rare case